x/lockup/keeper/lock_test.go

```diff
@@ -10,12 +10,218 @@ import (
+			expectedError:  types.ErrLockNotUnlocking,
+		},
+		{
+			name: "Invalid: Trying to rebond a synthetic's underlying lock",
+			setupFunctions: []func(){
+				lockTokensF,
+				startUnlockingF,
+				// create synthetic lockup
+				func() {
+					err := suite.App.LockupKeeper.CreateSyntheticLockup(suite.Ctx, defaultLockID, "synthetic", defaultDuration, false)
+					suite.Require().NoError(err)
+				},
+			},
+			rebondLockID:  defaultLockID,
+			expectedError: types.ErrLockHasSyntheticLockup,
+		},
+		{
+			name: "Invalid: Not the owner of a lock",
+			setupFunctions: []func(){
+				// lock is created not by default account
+				func() {
+					lockID := suite.LockTokens(arbitraryOwner, lockedCoins, defaultDuration)
+
+					suite.Require().Equal(defaultLockID, lockID)
+				},
+				startUnlockingF,
+			},
+			rebondLockID:  defaultLockID,
+			expectedError: types.ErrNotLockOwner,
+		},
+		{
+			name: "Invalid: unlock half of lock's supply - try to rebond initial lock",
+			setupFunctions: []func(){
+				lockTokensF,
+				// start unlocking half of the locked coins, then rebond the original lock
+				func() {
+					c := lockedCoins.Sub(sdk.NewCoins(sdk.NewInt64Coin("stake", lockedCoinsAmount/2)))
+					_, err := suite.App.LockupKeeper.BeginUnlock(suite.Ctx, defaultLockID, c)
+					suite.Require().NoError(err)
+				},
+			},
+			rebondLockID: defaultLockID, // it means we are trying to rebond a not-unbonding lock
+			// treat this kind of rebonding as if trying to rebond arbitrary not-unbonding lock
+			expectedError: types.ErrLockNotUnlocking,
+		},
+		{
+			name: "Valid: unlock half of lock's supply - try to rebond splitted lock",
+			setupFunctions: []func(){
+				lockTokensF,
+				// start unlocking half of the locked coins, then rebond the original lock
+				func() {
+					c := lockedCoins.Sub(sdk.NewCoins(sdk.NewInt64Coin("stake", lockedCoinsAmount/2)))
+					_, err := suite.App.LockupKeeper.BeginUnlock(suite.Ctx, defaultLockID, c)
+					suite.Require().NoError(err)
+				},
+			},
+			rebondLockID: defaultLockID + 1, // it means we are trying to rebond a splitted unbonding lock
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			for _, setupFunction := range tc.setupFunctions {
+				setupFunction()
+			}
+
+			// create two additional locks prior to testing to assert that they are not modified in the process
+			suite.LockTokens(noiseLocksOwner, lockedCoins, defaultDuration)
+			suite.LockTokens(noiseLocksOwner, lockedCoins, defaultDuration)
+			initialNoiseLocks := suite.App.LockupKeeper.GetAccountPeriodLocks(suite.Ctx, noiseLocksOwner)
+
+			// get initial testing lock
+			initialLock, err := suite.App.LockupKeeper.GetLockByID(suite.Ctx, defaultLockID)
+			suite.Require().NoError(err)
+
+			// calling end blocker to automatically remove matured locks (no-op if setup functions did not modify blocktime)
+			lockup.EndBlocker(suite.Ctx, *suite.App.LockupKeeper)
```

Comment:
note: this is needed for test cases that assert rebonding of unlocking lock at some stage of it's unlocking period (ex: half of unlocking time has elapsed, exactly the unlocking time has elapsed, unlocking time has elapsed in the past)